### ISSUE TITLE
Zellij crashes when clicking with mouse in pane

### ISSUE THREAD

**Issue by abannachZen** [zellij-org/zellij#816]
When starting a zellij session if I click in a pane zellij crashes. Even if the only pane available is the default.

![zellij-crash](https://user-images.githubusercontent.com/91550524/138894446-82cfab5a-89b0-45a8-96ad-e9949d36d9e1.gif)

If this is an unknown issue I can create a new thread for it.

`zellij 0.19.0`
`macOs Big Sur: 11.6`

Crash output
```
Error: thread 'stdin_handler' panicked at 'called `Option::unwrap()` on a `None` value': /Users/brew/Library/Caches/Homebrew/cargo_cache/registry/src/github.com-1ecc6299db9ec823/termion-1.5.6/src/event.rs:199
   0: backtrace::backtrace::trace
   1: backtrace::capture::Backtrace::new
   2: zellij_utils::errors::handle_panic
   3: std::panicking::rust_panic_with_hook
   4: std::panicking::begin_panic_handler::{{closure}}
   5: std::sys_common::backtrace::__rust_end_short_backtrace
   6: _rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::panicking::panic
   9: zellij_client::start_client::{{closure}}
  10: std::sys_common::backtrace::__rust_begin_short_backtrace
  11: core::ops::function::FnOnce::call_once{{vtable.shim}}
  12: std::sys::unix::thread::Thread::new::thread_start
  13: __pthread_start
```

_Originally posted by @abannachZen in https://github.com/zellij-org/zellij/issues/175#issuecomment-951972821_

**Comment by a-kenji** [comment]
@abannachZen 
Is that still an issue on the currently released version?

**Comment by a-kenji** [comment]
@stormcat24
Is that still an issue that you face on the currently released version?

**Comment by ozsay** [comment]
hey, this issue is still happening in the last version

```
Originating Thread(s):

Error: thread 'stdin_handler' panicked at 'called `Option::unwrap()` on a `None` value': /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/termion-1.5.6/src/event.rs:193
   0: __mh_execute_header
   1: __mh_execute_header
   2: __mh_execute_header
   3: __mh_execute_header
   4: __mh_execute_header
   5: __mh_execute_header
   6: __mh_execute_header
   7: __mh_execute_header
   8: __mh_execute_header
   9: __mh_execute_header
  10: __mh_execute_header
  11: __mh_execute_header
  12: __mh_execute_header
  13: __pthread_start
```

**Comment by tlinford** [comment]
@imsnif looks like it's coming from termion, seems like it's getting a csi sequence that's tripping it up.
@ozsay what terminal emulator and os is this happening on?

**Comment by ozsay** [comment]
@tlinford I'm on macos.

Tested it on: iTerm, Alacritty and the mac Terminal app. it happens everywhere

**Comment by imsnif** [comment]
@tlinford - do we have a way of finding out which sequence caused this? Maybe with a debug version?

**Comment by ozsay** [comment]
not sure if it is, but after the crash, the shell is left with some characters I suppose are a sequence ?

```shell
Originating Thread(s):

Error: thread 'stdin_handler' panicked at 'called `Option::unwrap()` on a `None` value': /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/termion-1.5.6/src/event.rs:193
   0: __mh_execute_header
   1: __mh_execute_header
   2: __mh_execute_header
   3: __mh_execute_header
   4: __mh_execute_header
   5: __mh_execute_header
   6: __mh_execute_header
   7: __mh_execute_header
   8: __mh_execute_header
   9: __mh_execute_header
  10: __mh_execute_header
  11: __mh_execute_header
  12: __mh_execute_header
  13: __pthread_start
  
> 0;71;17m
```

not sure if it's what you're looking for, but it's different each time.. some examples: `0;69;19M` `0;48;17M` `0;42;20m`

Interestingly though, rarely, `zellij` is not crashing on the first mouse click but display the following.

![](https://user-images.githubusercontent.com/3464930/155089752-44aef1cc-54b8-4191-bf39-e845573c4f09.png)

the next click is crashing with these characters `;63;15m`

**Comment by tlinford** [comment]
@ozsay that's very helpful thanks.
Those characters are encoded mouse buttons and coordinates, seems like for some reason the mouse instructions are getting split up and aren't parsed correctly.

Would you be up to running a debug build off of a debug branch that I can prepare with some  extra logging of what's arriving from stdin?

**Comment by ozsay** [comment]
sure let's set it up

**Comment by tlinford** [comment]
@ozsay awesome, thanks for helping out. Branch is `debug-mouse-crash` (https://github.com/zellij-org/zellij/tree/debug-mouse-crash).

It would be best to start out with clean logs, so please clear the logs in `/tmp/zellij-1000`, after that you can start up with `cargo make run`, reproduce the crash, and get the `zellij.log` file.

Please note that this is going to log anything you input, so avoid writing anything sensitive :slightly_smiling_face:

**Comment by ozsay** [comment]
```
INFO   |zellij_client            | 2022-02-22 12:20:40.292 [main      ] [zellij-client/src/lib.rs:118]: Starting Zellij client!
INFO   |zellij_server            | 2022-02-22 12:20:40.319 [main      ] [zellij-server/src/lib.rs:186]: Starting Zellij server!
INFO   |zellij_server::wasm_vm   | 2022-02-22 12:20:40.380 [wasm      ] [zellij-server/src/wasm_vm.rs:86]: Wasm main thread starts
INFO   |zellij_client::stdin_hand| 2022-02-22 12:20:46.221 [stdin_handler] [zellij-client/src/stdin_handler.rs:66]: read from stdin: [27, 91, 60]
INFO   |zellij_server::wasm_vm   | 2022-02-22 12:20:47.646 [wasm      ] [zellij-server/src/wasm_vm.rs:226]: wasm main thread exits
```

**Comment by tlinford** [comment]
@ozsay I just added a push to hopefully work around this problem, could you give it a try again and let me know (logfile again would also be helpful).

**Comment by ozsay** [comment]
it happens again, but less frequently as you can see in the logs

```
INFO   |zellij_client            | 2022-02-22 13:58:24.097 [main      ] [zellij-client/src/lib.rs:118]: Starting Zellij client!
INFO   |zellij_server            | 2022-02-22 13:58:24.124 [main      ] [zellij-server/src/lib.rs:186]: Starting Zellij server!
INFO   |zellij_server::wasm_vm   | 2022-02-22 13:58:24.183 [wasm      ] [zellij-server/src/wasm_vm.rs:86]: Wasm main thread starts
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.773 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.774 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Unsupported([27, 91])
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.774 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [60, 48, 59, 55, 53, 59, 49, 53, 77]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.774 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('<'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.774 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('0'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.775 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.775 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('7'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.775 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.775 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.776 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('1'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.776 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.776 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('M'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.919 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91, 60, 48, 59, 55, 53, 59, 49, 53, 109]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:30.919 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Mouse(Release(75, 15))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.739 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.739 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Unsupported([27, 91])
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.740 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [60, 48, 59, 55, 53, 59, 49, 53, 77]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.740 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('<'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.740 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('0'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.741 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.741 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('7'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.741 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.741 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.742 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('1'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.742 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.742 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('M'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.862 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Unsupported([27, 91])
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [60, 48, 59, 55, 53, 59, 49, 53, 109]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('<'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('0'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('7'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.863 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.864 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char(';'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.864 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('1'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.864 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('5'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:31.864 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Key(Char('m'))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.572 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91, 60, 48, 59, 55, 53, 59, 49, 53, 77]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.572 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Mouse(Press(Left, 75, 15))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.729 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91, 60]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.730 [stdin_handler] [zellij-client/src/stdin_handler.rs:71]: found mouse sequence start
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.730 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [48, 59, 55, 53, 59, 49, 53, 109]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.730 [stdin_handler] [zellij-client/src/stdin_handler.rs:77]: completing mouse sequence
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:32.730 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Mouse(Release(75, 15))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.626 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91, 60]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.626 [stdin_handler] [zellij-client/src/stdin_handler.rs:71]: found mouse sequence start
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.626 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [48, 59, 55, 53, 59, 49, 53, 77]
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.626 [stdin_handler] [zellij-client/src/stdin_handler.rs:77]: completing mouse sequence
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.626 [stdin_handler] [zellij-client/src/stdin_handler.rs:142]: event: Mouse(Press(Left, 75, 15))
INFO   |zellij_client::stdin_hand| 2022-02-22 13:58:33.787 [stdin_handler] [zellij-client/src/stdin_handler.rs:68]: read from stdin: [27, 91, 60, 48]
INFO   |zellij_server::wasm_vm   | 2022-02-22 13:58:35.155 [wasm      ] [zellij-server/src/wasm_vm.rs:226]: wasm main thread exits
```

**Comment by tlinford** [comment]
thanks again @ozsay, the logs are very helpful.

**Comment by tlinford** [comment]
@ozsay i made another change on the branch, would be great if you could test it out again.

**Comment by ozsay** [comment]
looks like that did it :) no more crashes

**Comment by tlinford** [comment]
awesome! this change for now breaks bracketed paste, so it may take a while to get this fully sorted out.

**Comment by ozsay** [comment]
no worries,

I'm still trying to see how to fully migrate to `zellij` and `alacritty` (using terminal tabs ATM in item2).

one thing I saw (will probably open an issue), is sometimes commands stdout is flickering when running in `zellij`

**Comment by imsnif** [comment]
This was fixed in 0.26.0
